cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/selector/BaseStorageSelector.java

```diff
@@ -0,0 +1,15 @@
+package com.linkedin.openhouse.cluster.storage.selector;
+
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/** An abstract class for all Storage selectors */
+public abstract class BaseStorageSelector implements StorageSelector {
+
+  @Autowired private StorageProperties storageProperties;
+
+  @Override
+  public String getName() {
+    return storageProperties.getStorageSelector().getName();
```

Comment:
This should return the overriden class's classname for example: if `DefaultStorageSelector extends BaseStorageSelector`, then the name should be `DefaultStorageSelector` can you check how we can do that in Java? maybe `this.getClass.getShortName`? if its not possible, lets implement it in respective storageSelector